crates/circuit/src/circuit_data.rs

```diff
@@ -995,6 +996,24 @@ impl CircuitData {
         self.param_table.clear();
     }
 
+    /// Counts the number of times each operation is used in the circuit.
+    ///
+    /// # Parameters
+    /// - `self` - A mutable reference to the CircuitData struct.
+    ///
+    /// # Returns
+    /// An IndexMap containing the operation names as keys and their respective counts as values.
+    pub fn count_ops(&mut self) -> IndexMap<String, usize, ::ahash::RandomState> {
+        let mut ops_count: IndexMap<String, usize, ::ahash::RandomState> = IndexMap::default();
+        for instruction in &self.data {
+            *ops_count
+                .entry(instruction.op.view().name().to_string())
```

Comment:
You don't actually need to build the `OperationRef` view here, `PackedOperation` has a name method you can just call. This is actually what I think blocked you from returning a `&str` as the key before.

```suggestion
                .entry(instruction.op.name().to_string())
```

Reply:
Ah for some reason I didn't see that and that made things more complicated for me